core/creator/src/main/java/io/quarkus/creator/phase/nativeimage/NativeImagePhase.java

```diff
@@ -407,7 +407,7 @@ public void provideOutcome(AppCreator ctx) throws AppCreatorException {
             if (additionalBuildArgs != null) {
                 command.addAll(additionalBuildArgs);
             }
-            command.add("-H:InitialCollectionPolicy='com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime'"); //the default collection policy results in full GC's 50% of the time
+            command.add("-H:InitialCollectionPolicy=\"com.oracle.svm.core.genscavenge.CollectionPolicy\\$BySpaceAndTime\""); //the default collection policy results in full GC's 50% of the time
```

Comment:
This is not necessary. These arguments are used by `ProcessBuilder` which does not use a shell to execute commands.

Reply:
Since the problem is related to Docker, maybe there should be a second escaping pass when docker is used?